test/scripttest/logging_test.go

```diff
@@ -57,47 +67,177 @@ func TestLogging(t *testing.T) {
+//
+// The test task will then:
+//  1. Print the message to log
+//  2. Ping the test-server and block waiting for the reply
+//  3. Print the test-server reply, exiting successfully if response is 200.
+//
+// By greping the final log for response payload we ensure that our test-server
+// did indeed reply to the test task. By not replying from the test-server
+// until we've read the message from the livelog, we ensure that the livelog
+// works. If livelog doesn't work this should deadlock, or request should
+// timeout in the test task, instead of returning 200.
+//
+// To increase test coverage we shall try with a few different message sizes
+// and behaviors for the test-server once it has replied. The behaviors
+// ensures that a blocking livelog reader doesn't block the worker, or cause
+// other undesired behavior in the worker.
+func TestLiveLog(t *testing.T) {
 	if goruntime.GOOS == "windows" {
 		// On windows this fails with:
 		// dial tcp 127.0.0.1:49523: socket: The requested service provider could not be loaded or initialized.
 		// Calling the URL from this process works, but the subprocess fails...
 		t.Skip("TODO: Fix the GetUrl on windows, probably firewall interference")
 	}
 
-	var u string
+	// Generate some messages that can be written log, so we can read them from livelog
+	// Trying different message sizes tests the buffer slicing code a bit.
+	randMessage := make([]byte, 1024*8+27)
+	rand.Read(randMessage)
+	messages := map[string]string{
+		"special":   "@", // single special character
+		"short":     "hello world, this is a slightly longer message, maybe that tests something more",
+		"very-long": base64.StdEncoding.EncodeToString(randMessage),
+	}
+
+	// Generate some replies that can be written log, so we know that test-server was called
+	randReply := make([]byte, 1024*8+27)
+	rand.Read(randReply)
+	replies := map[string]string{
+		"special":   "*", // single special character
+		"short":     "magic-words, this is a slightly longer reply, maybe that tests something more",
+		"very-long": base64.StdEncoding.EncodeToString(randReply),
+	}
+
+	done := make(chan struct{})
+	var s *httptest.Server
 	workertest.Case{
 		Engine:       "script",
-		Concurrency:  1,
+		Concurrency:  5, // this test runs a lot of task, we better run them in parallel
 		EngineConfig: engineConfig,
 		PluginConfig: pluginConfig,
 		Setup: func(t *testing.T, env workertest.Environment) func() {
-			// Create test server with magic-words, we'll grep for later
-			s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// Create test-server with magic-words, we'll grep for later
+			var wg sync.WaitGroup
+			s = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				taskID := r.URL.Query().Get("taskId")
+				msg := messages[r.URL.Query().Get("msgSize")]
+				behavior := r.URL.Query().Get("behavior")
+				reply := replies[r.URL.Query().Get("replySize")]
```

Comment:
Are `msg` and `reply` buffers or buffer sizes?

Reply:
They are strings... `messages` is a mapping from _string representing size_ to _message as string_.